index/field/doc_id_multi_field_intersect_iterator.go

```diff
@@ -0,0 +1,99 @@
+package field
+
+import (
+	"github.com/xichen2020/eventdb/document/field"
+	"github.com/xichen2020/eventdb/index"
+)
+
+// DocIDMultiFieldIntersectIterator is an iterator that iterates over multiple
+// fields for values associated with documents in both the given doc ID set
+// iterator and the multi field iterator.
+type DocIDMultiFieldIntersectIterator struct {
+	docIt        index.DocIDSetIterator
+	multiFieldIt *MultiFieldIterator
```

Comment:
nit: `MultiFieldIterator` wasn't implemented in this PR but should it be called `MultiFieldIntersectIterator` instead?

Reply:
Sure sounds good.